sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -351,6 +351,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DYNAMIC_PARTITION_PRUNING_MAX_LENGTH =
+    buildConf("spark.sql.optimizer.dynamicPartitionPruning.max.length")
+      .doc("The maximum character length of the plan for the nested subquery.When the set length " +
+        "is too large, it may cause the final plan to exceed the limit or memory overflow.")
+      .version("3.3.0")
```

Comment:
Let's make it 4.0.0

Reply:
Your suggestion is very good. I will modify the relevant content and submit it again. Do you have any better suggestions for the modified code?

Reply:
The adapted version has been changed from 3.3.0 to 4.0.0.